File: tests/rosetta/transpiler/php/dining-philosophers-2.php
<?php
ini_set('memory_limit', '-1');
function main() {
  $philosophers = ['Aristotle', 'Kant', 'Spinoza', 'Marx', 'Russell'];
  $hunger = 3;
  echo rtrim('table empty'), PHP_EOL;
  foreach ($philosophers as $p) {
  echo rtrim($p . ' seated'), PHP_EOL;
};
  $i = 0;
  while ($i < count($philosophers)) {
  $name = $philosophers[$i];
  $h = 0;
  while ($h < $hunger) {
  echo rtrim($name . ' hungry'), PHP_EOL;
  echo rtrim($name . ' eating'), PHP_EOL;
  echo rtrim($name . ' thinking'), PHP_EOL;
  $h = $h + 1;
};
  echo rtrim($name . ' satisfied'), PHP_EOL;
  echo rtrim($name . ' left the table'), PHP_EOL;
  $i = $i + 1;
};
  echo rtrim('table empty'), PHP_EOL;
}
main();

File: tests/rosetta/transpiler/Java/dining-philosophers-2.java
public class Main {

    static void main() {
        String[] philosophers = new String[]{"Aristotle", "Kant", "Spinoza", "Marx", "Russell"};
        int hunger = 3;
        System.out.println("table empty");
        for (String p : philosophers) {
            System.out.println(p + " seated");
        }
        int i = 0;
        while (i < philosophers.length) {
            String name = philosophers[i];
            int h = 0;
            while (h < hunger) {
                System.out.println(name + " hungry");
                System.out.println(name + " eating");
                System.out.println(name + " thinking");
                h = h + 1;
            }
            System.out.println(name + " satisfied");
            System.out.println(name + " left the table");
            i = i + 1;
        }
        System.out.println("table empty");
    }
    public static void main(String[] args) {
        main();
    }
}
